pkg/cli/rm_helper.go

```diff
@@ -150,12 +151,31 @@ func removeApp(arg string, c client.Client, cmd *cobra.Command, force bool) erro
 		}
 	}
 	app, err := c.AppDelete(cmd.Context(), arg)
-
 	if err != nil {
 		return fmt.Errorf("deleting app %s: %w", arg, err)
 	}
+
 	if app != nil {
-		fmt.Println("Removed: " + arg)
+		// Ensure this gets printed whether we wait or not
+		defer func() {
+			if err == nil || apierrors.IsNotFound(err) {
+				fmt.Println("Removed: " + arg)
+			}
+		}()
+		if wait {
+			fmt.Printf("Waiting for app %s to be removed...\n", arg)
+			for {
+				select {
+				case <-cmd.Context().Done():
+					return cmd.Context().Err()
+				default:
+					if _, err = c.AppGet(cmd.Context(), arg); apierrors.IsNotFound(err) {
+						return nil
+					}
```

Comment:
I think we also want to return an error -- or at least print one --  when `err != nil && !err.IsNotFound(err)`; i.e. any error other than NOT_FOUND.

Reply:
Logging at debug level.